modules/storage/cosmos-db/main.bicep

```
@@ -1,99 +1,353 @@
-@description('Deployment Location')
-param location string
+@description('Deployment region name. Default is the location of the resource group.')
+param location string = resourceGroup().location
 
-@description('Prefix of Cosmos DB Resource Name')
-param prefix string = enableCassandra ? 'coscas' : 'cosmos'
+@description('The bakend API type of Cosmos DB database account. The API selection cannot be changed after account creation. Possible values: "cassandra", "gremlin", "mongodb", "sql", "table".')
+@allowed([ 'cassandra', 'gremlin', 'mongodb', 'sql', 'table' ])
+param backendApi string = 'cassandra'
 
-@description('Name of Cosmos DB Resource')
-param name string = '${prefix}${uniqueString(resourceGroup().id, location)}'
+@description('The name of the Cosmos DB account. Character limit: 3-44, valid characters: lowercase letters, numbers, and hyphens. It must me unique across Azure.')
+param name string = 'cdb${backendApi}-${uniqueString(resourceGroup().id, location)}'
```

```suggestion
@description('Prefix of Cosmos DB Resource Name. Not used if name is provided.')
param prefix string = {
  cassandra: 'coscas'
  gremlin: 'cosgrm'
  mongodb: 'cosmon'
  sql: 'cosmos'
  table: 'costab'
}[backendApi]

param name string = '${prefix}${uniqueString(resourceGroup().id, location)}'
```

Let's add back the prefix, and following the abbreviations documented [here](https://learn.microsoft.com/en-us/azure/cloud-adoption-framework/ready/azure-best-practices/resource-abbreviations#databases)
Then, in your test cases you can use `prefix: test01-` instead of `name: 'test01-${uniqueName}'`

---

Added prefix as suggested, min and max is only applicable for `int` type param so not added.